### PR TITLE
Use Firestore Listener to Update Feed and Approval Screens

### PR DESCRIPTION
**Pull request by alliu879**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
The main goal of this PR is to remove clunky state hooks used for updating job lists in both feed and approval screens and instead use `react-firestore-listener` package to listen to changes to both `approvedJobs` and `notApprovedJobs` collections and update UI accordingly. Note that while this greatly simplifies the codebase, there are two main considerations:

1. The hook used by the firestore listener gives us a list of `Partial<Job>` objects, each of which contains extraneous firestore-related metadata that causes errors when we attempt to approve a pending `JobCard`. To resolve this, a new function is created to parse the `Partial<Job>` object (`parseFirestoreListenerJob`) before passing it in to be approved. The new function does this using `jobInstance`, which is an implementation of the `Job` interface. This implementation can also be used reduce the use of hard-coded job field values in future PRs (i.e. `date`, `phone`, `probationPeriod`, etc.)
2. The original design for displaying a list of `JobCards` used the `getAllJobs` function, which read all jobs from firebase and then parsed them to give each job an `id` field. This `id` field is necessary to correctly delete/approve pending jobs — however, it is not longer correctly populated when we use the firestore listener to get all jobs instead. To resolve this, we populate the `id` field immediately after the job has been created with the document id of the pending job.

Also included in this PR:
- Fix missing `key` prop error for items in list
- Basic refactoring of variables/functions to be more descriptive i.e. `filteredList` -> `filterApprovedJobs`

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
n/a

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Test that functionality for creating pending and approved jobs work as expected.
- Ensure that there is no lag in the updating of pending and approved job lists

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
- [react-firestore-listener](https://www.npmjs.com/package/react-firestore-listener)

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
This PR is an updated version of the following PR reviewed by @wangannie: https://github.com/calblueprint/chinese-newcomers/pull/72  

CC: @surajr10 @themiachou8

### CODE REVIEW

File: src/firebase/firestore/job.ts [hunk 1]
```diff
@@ -62,6 +62,13 @@ export const getMonthlyCounter = async (): Promise<number> => {
   return data?.monthlyCounter;
 };
 
+function parseFirestoreListenerJob(jobId: string, job: Partial<Job>) {
+  const jobKeys = Object.keys(jobInstance);
+  const parsedJob = Object.fromEntries(jobKeys.map(k => [k, job[k as keyof typeof job]]));
```
Comment (surajr10) [inline src/firebase/firestore/job.ts:67]: is the "as keyof typeof" part necessary?

**Review by surajr10 — APPROVED** [review]
lgtm!

File: src/firebase/firestore/job.ts (hunk 1 above)
Comment (alliu879) [reply src/firebase/firestore/job.ts:67]: yep without typing this we will get eslint typescript errors
